modules/exploits/linux/samba/is_known_pipename.rb

```diff
@@ -209,7 +209,7 @@ def find_writeable_share_path
     end
 
     share_info.each do |share|
-      next if share.first.upcase == 'IPC$'
+      next if share&.first&.upcase == 'IPC$'
```

Comment:
I think this is little bit more clear:
```suggestion
      next if share.first.blank?
      next if share.first.upcase == 'IPC$'
```
Also, is the bug reproducible against functional instance or is it caused by misconfigured instance/incorrect communication?

Reply:
This bug has never appeared to me. I've just seen the issue and added logic to check if the value's nil or not.